tdms/include/grid_labels.h

```diff
@@ -11,11 +11,20 @@
  */
 class GridLabels{
 public:
-    double *x = nullptr;  // Start of the labels in the x direction
-    double *y = nullptr;  //                            y
-    double *z = nullptr;  //                            z
+    double *x = nullptr;  //< Start of the labels in the x direction
+    double *y = nullptr;  //< Start of the labels in the y direction
+    double *z = nullptr;  //< Start of the labels in the z direction
 
     GridLabels() = default;
 
     explicit GridLabels(const mxArray *ptr);
+
+    /**
+     * @brief Set values by copying from another GridLabels object
+     *
+     * @param other_labels The GridLabels object to copy values from
+     * @param i_l,j_l,k_l The first item to copy from the x, y, z attributes (respectively)
+     * @param i_u,j_u,k_u The final (inclusive) item to copy from the x, y, z attributes (respectively)
+     */
+    void initialise_from(GridLabels &labels_to_copy_from, int i_l, int i_u, int j_l, int j_u, int k_l, int k_u);
```

Comment:
Initialise != constructor?
(Just asking) why we don't have a `GridLabels` constructor?

Reply:
That's a question for @t-young31 if he remembers :sweat_smile:
 
Although I think it's just because we never actually write our own labels, just cherry-pick the relevant ones from what MATLAB/our input gives us.